File: 520-detect-capital/solution.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        int count = 0;
        for (int i = 0; i < word.size(); i ++) {
            if (isupper(word[i])) count ++;
        }
        return (count == 0) || 
               (isupper(word[0]) && (count == 1)) || 
               count == word.size();
    }
};

File: 520-detect-capital/solution.py
class Solution(object):
    def detectCapitalUse(self, word):
        """
        :type word: str
        :rtype: bool
        """
        # if all([w.isupper() for w in word]):
        #     return True
        # first_cap = False
        # for i, w in enumerate(word):
        #     if word[i].isupper() and i == 0:
        #         first_cap = True
        #     elif word[i].isupper() and i != 0:
        #         return False
        # return True
        return word.isupper() or word.islower() or word.istitle()